File: HomeWork/Home3.2/src/main/java/com/example/Triangle.java
package com.example;

public class Triangle {
    private double sideA;
    private double sideB;
    private double sideC;

    public Triangle(double sideA, double sideB, double sideC) {
        this.sideA = sideA;
        this.sideB = sideB;
        this.sideC = sideC;
    }

    public double getHeight(char side) {
        switch (side) {
            case 'a':
                return (2 * area()) / sideA;
            case 'b':
                return (2 * area()) / sideB;
            case 'c':
                return (2 * area()) / sideC;
            default:
                throw new IllegalArgumentException("Invalid side: " + side);
        }
    }

    private double area() {
        double s = (sideA + sideB + sideC) / 2;
        return Math.sqrt(s * (s - sideA) * (s - sideB) * (s - sideC));
    }
}

File: Class/practice3.2/Triangle.cpp
#include "UnitTest.hpp"
#include <cmath>
#include <functional> // Add this for std::function
#include <vector>     // Add this for std::vector

class Triangle {
public:
    Triangle(double a, double b, double c) : sideA(a), sideB(b), sideC(c) {}

    double calculateArea() const {
        // Heron's formula to calculate the area of a triangle
        double s = (sideA + sideB + sideC) / 2;
        return std::sqrt(s * (s - sideA) * (s - sideB) * (s - sideC));
    }

private:
    double sideA, sideB, sideC;
};

int main() {
    UnitTests tests; // Corrected class name

    // Test case for Triangle class
    tests.addTest("TriangleAreaTest", [](){
        Triangle triangle(3, 4, 5);
        ASSERT_EQ(triangle.calculateArea(), 6.0);
    });

    // Run the tests
    tests.run();

    return 0;
}
